vmdk_plugin/drivers/vmdk/vmdkops/vmdkops.go

```diff
@@ -59,14 +60,32 @@ func (v VmdkOps) Remove(name string, opts map[string]string) error {
 	return err
 }
 
-// Attach a volume
-func (v VmdkOps) Attach(name string, opts map[string]string) ([]byte, error) {
+// RawAttach attaches a volume and returns a raw response.
+func (v VmdkOps) RawAttach(name string, opts map[string]string) ([]byte, error) {
 	log.Debugf("vmdkOps.Attach name=%s", name)
 	str, err := v.Cmd.Run("attach", name, opts)
 	if err != nil {
+		log.WithFields(log.Fields{"name": name, "opts": opts, "error": err}).Error("RawAttach failed ")
 		return nil, err
 	}
-	return str, err
+	return str, nil
+}
+
+// Attach attaches a volume and returns the disk's VolumeDevSpec.
+func (v VmdkOps) Attach(name string, opts map[string]string) (*fs.VolumeDevSpec, error) {
+	str, err := v.RawAttach(name, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	var volDev fs.VolumeDevSpec
+	err = json.Unmarshal(str, &volDev)
+	if err != nil {
+		log.WithFields(log.Fields{"name": name, "opts": opts, "bytes": str,
+			"error": err}).Error("Failed to unmarshal ")
+		return nil, err
```

Comment:
If for some reason this json decode fails then is the device going to be left attached to the VM? The raw-attach itself succeeded  so the volume is attached to the VM at this point.